test/acceptance/post_test.exs

```diff
@@ -0,0 +1,263 @@
+defmodule Test.Acceptance.PostTest do
+  use ExUnit.Case, async: true
+
+  defmodule Author do
+    use Ash.Resource,
+      data_layer: Ash.DataLayer.Ets,
+      extensions: [
+        AshJsonApi.Resource
+      ]
+
+    ets do
+      private?(true)
+    end
+
+    json_api do
+      type("author")
+
+      routes do
+        base("/authors")
+        get(:default)
+        index(:default)
+      end
+
+      fields [:id, :name]
+    end
+
+    actions do
+      read(:default)
+
+      create(:default)
+    end
+
+    attributes do
+      attribute(:id, :uuid, primary_key?: true)
+      attribute(:name, :string)
+    end
+
+    relationships do
+      has_many(:posts, Test.Acceptance.PostTest.Post, destination_field: :author_id)
+    end
+  end
+
+  defmodule Post do
+    use Ash.Resource,
+      data_layer: Ash.DataLayer.Ets,
+      extensions: [
+        AshJsonApi.Resource
+      ]
+
+    ets do
+      private?(true)
+    end
+
+    json_api do
+      type("post")
+
+      routes do
+        base("/posts")
+        get(:default)
+        index(:default)
+        post(:default)
+      end
+
+      fields [:id, :name, :hidden, :email, :author]
+    end
+
+    actions do
+      read(:default)
+
+      create :default do
+        accept([:id, :name, :hidden, :author])
+      end
+    end
+
+    attributes do
+      attribute(:id, :uuid, primary_key?: true)
+      attribute(:name, :string)
+      attribute(:hidden, :string)
+
+      attribute(:email, :string,
+        allow_nil?: true,
+        constraints: [
+          match: ~r/[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}/
+        ]
+      )
+    end
+
+    relationships do
+      belongs_to(:author, Test.Acceptance.PostTest.Author)
+    end
+  end
+
+  defmodule Api do
+    use Ash.Api,
+      extensions: [
+        AshJsonApi.Api
+      ]
+
+    json_api do
+      log_errors?(false)
+    end
+
+    resources do
+      resource(Post)
+      resource(Author)
+    end
+  end
+
+  import AshJsonApi.Test
+
+  @tag :attributes
+  describe "invalid_post" do
+    test "create without all attributes in accept list" do
+      # dummy response.
```

Comment:
do we still need this?

Reply:
no, it was just for comparison. removing it.